### PR TITLE
BSM-132: Cannot Edit an Edge after creation

### PR DESCRIPTION
**Pull request by christianpape**
Jira: http://issues.opennms.org/browse/BSM-132
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS462

### CODE REVIEW

File: features/bsm/vaadin-adminpage/src/main/java/org/opennms/netmgt/bsm/vaadin/adminpage/BusinessServiceEdgeEditWindow.java [hunk 1]
```diff
@@ -312,6 +318,37 @@ public BusinessServiceEdgeEditWindow(final BusinessService businessService,
...
+                    }
+
+                    case REDUCTION_KEY: {
+                        Set<ReductionKeyEdge> edges = businessService.getReductionKeyEdges();
+                        edges.remove(edge);
+                        businessService.setReductionKeyEdges(edges);
+                        break;
+                    }
+                }
+            }
```
Comment (mvrueden) [inline features/bsm/vaadin-adminpage/src/main/java/org/opennms/netmgt/bsm/vaadin/adminpage/BusinessServiceEdgeEditWindow.java]: Can we move this to the BusinessServiceManager or BusinessService object? 
How about creating a `getEdges(EdgeType)` and `setEdges(EdgeType, Collection)` method?

**Comment by mvrueden** [comment]
Please also provide a smoke test to verify that you really can edit an edge.
I see two test cases:
1. edit a already persisted edge
2. edit a transient (non persisted) edge
